tests/testthat/test-standalone-checks.R

```diff
@@ -138,19 +138,18 @@ test_that("check functions work", {
 
   # check_scalar_range()
   expect_silent(check_scalar_range(5, c(1, 20)))
-  # expect_silent(check_scalar_range(5, c(1, 5), include_bounds = c(FALSE, TRUE)))
+  expect_silent(check_scalar_range(5, c(1, 5), include_bounds = c(FALSE, TRUE)))
 
   expect_error(
     check_scalar_range(5, c(0, 1)),
     "The `5` argument must be in the interval `(0, 1)` and length 1.",
     fixed = TRUE
   )
-  # expect_error(
-  #   check_scalar_range(c(4, 5), c(1, 5), include_bounds = c(TRUE, FALSE)),
-  #   'The `5` argument must be in the interval `[1, 5)`.',
-  #   fixed = TRUE
-  # )
-  # parent.env(call) <- parent.frame()
+  expect_error(
+    check_scalar_range(c(4, 5), c(1, 5), include_bounds = c(TRUE, FALSE)),
+    'The `5` argument must be in the interval `[1, 5)`.',
+    fixed = TRUE
+  )
```

Comment:
Hi @ddsjoberg,

This test is currently failing - many of the check functions call other check functions and pass through their more specific error messages, but the innermost functions don't carry the environment variables from the outermost functions if they are not also arguments of that function. Thus when `cli::cli_abort()` is ulitmately called within the innermost check function, any variables used in the specific error message may not be present (they are present in `call` but `cli::cli_abort()` gets its variables from the `.env` argument, not `call`). 

E.g. In this case `check_scalar_range()` has argument `include_bounds` and passes through the following message:
 ```
"The {.arg {arg_name}} argument must be in the interval {.code {ifelse(include_bounds[1], '[', '(')}{range[1]}, {range[2]}{ifelse(include_bounds[2], ']', ')')}} and length {.val {1}}."
``` 

The function calls `check_scalar()`, which then calls `check_length()` which prints the error message. But `include_bounds` is not found by `check_length`, so the following error occurs when trying to formulate the message:

``` r
Error in `fun(..., .envir = .envir)`:
! Could not evaluate cli `{}` expression: `ifelse(include_boun…`.
Caused by error in `eval(expr, envir = envir)`:
! object 'include_bounds' not found
```

How do you want to handle cases like this? There's not many (if any other than this one), but stil a possibility that we should probably account for.

Reply:
Ahhhh, that makes sense! Oof!

My first inclination is to add an `env` argument that we would evaluate the `cli::cli_abort()` in? I think there are some details that need to be worked out if we went that route.

Have you thought about it? What do you think?

Reply:
I think I remedied the issue in this PR https://github.com/ddsjoberg/standalone/pull/15

Reply:
Your solution is great!! Very elegantly done